hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/ClusterMapReduceTestCase.java

```diff
@@ -43,8 +45,21 @@
  * The DFS filesystem is formated before the testcase starts and after it ends.
  */
 public abstract class ClusterMapReduceTestCase {
+  private static final String TEST_ROOT_DEFAULT_PATH =
+      System.getProperty("test.build.data", "target/test-dir");
```

Comment:
Ahh. I see it now.  They are different.  But I don't think you need to call getProperty.  Just pass "target/test-dir" as the default for setupTestRootDir.

Reply:
The point that if the property "test.build.data" is set globally to the unit tests, then the `ClusterMapReduceTestCase.java` should also reads this value. Otherwise, all unit tests would not consume the global properties set to the JVM. It will be a mess in the folders